apps/mobile/src/features/settings/theme-modal.tsx

```diff
@@ -1,21 +1,67 @@
 import { RefObject } from 'react';
 import { useSharedValue } from 'react-native-reanimated';
+import { useSafeAreaInsets } from 'react-native-safe-area-context';
 
 import { CLOSED_ANIMATED_SHARED_VALUE, Modal } from '@/components/bottom-sheet-modal';
+import { ModalHeader } from '@/components/headers/modal-header';
+import { useToastContext } from '@/components/toast/toast-context';
+import { ThemeStore, useSettings } from '@/state/settings/settings.slice';
 import { BottomSheetModal } from '@gorhom/bottom-sheet';
 import { t } from '@lingui/macro';
+import { useTheme } from '@shopify/restyle';
 
-import { Text } from '@leather.io/ui/native';
+import { Box, Cell, SunIcon, Theme } from '@leather.io/ui/native';
 
 interface ThemeModalProps {
   modalRef: RefObject<BottomSheetModal>;
 }
 export function ThemeModal({ modalRef }: ThemeModalProps) {
   const animatedIndex = useSharedValue<number>(CLOSED_ANIMATED_SHARED_VALUE);
+  const { bottom } = useSafeAreaInsets();
+  const theme = useTheme<Theme>();
+  const settings = useSettings();
+  const { displayToast } = useToastContext();
+
+  function onPressUpdateTheme(theme: ThemeStore) {
+    settings.changeTheme(theme);
+    displayToast({ title: t`Theme updated`, type: 'success' });
```

Comment:
Do we need toast here? Will be pretty obvious that the theme has changed

Reply:
@mica000 it is shown in the design, what do you think?

Reply:
Good point, probably not but it's mostly added to keep consistently with all the other areas of the settings. Wouldn’t be weird to have all of others showing a toaster and only this one not?

Reply:
Although, we might want to keep as there is also a case where users wont be able to see the change which is when you switch to the same theme the system is already using.